### PR TITLE
Feature/login prompt on like button click

### PR DESCRIPTION
**Pull request by tsrahm**
It would be easier to use route params with the auth0 redirectUri if possible. Right now, that fails because the url is not whitelisted.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -175,7 +171,7 @@ class App extends React.Component {
             this.handleAuthentication(props);
             return <Callback {...props} />
           }}/>
-          <Route path="/login" render={(props) => <Login auth={auth} {...props} />} />
+          <Route path="/login/:itemId/:definitionId" render={(props) => <Login auth={auth} {...props} />} />
```
Comment (nlkluth) [inline src/App.js]: does it make sense to use a queryParam so we can use the same route with path="/login"?

File: src/utils/Auth.js [hunk 2]
```diff
@@ -25,25 +26,41 @@ export default class Auth {
     this.auth0.authorize();
   }
 
-  handleAuthentication(callback) {
+  loginAndLike(itemId, definitionId) {
+    const params = (itemId && definitionId) ? `?itemId=${itemId}&definitionId=${definitionId}` : '';
```
Comment (nlkluth) [inline src/utils/Auth.js]: i like that you use a queryparam here on the callback to auth0. I wonder if we can make it simple and just add our queryparams to the auth0 callback every time without conditionally checking

File: src/AcronymLike.js [hunk 3]
```diff
@@ -32,7 +37,7 @@ const AcronymLike = (props) => {
       >
         <IconButton
           tooltip="Like"
-          onClick={() => clickHandler(isAuthenticated, liked, like, itemId, definitionId)}
+          onClick={() => clickHandler(like, definitionId, itemId, liked, isAuthenticated)}
```
Comment (NFabrizio) [inline src/AcronymLike.js]: Just curious. Why did you change the order of these arguments?

File: src/Login.js [hunk 4]
```diff
@@ -1,10 +1,17 @@
...
-const Login = ({ auth }) => (
-  <div>
-    <p> You need to be logged in to do that</p>
-    <button onClick={auth.login}>Login</button>
-  </div>
-);
+const Login = ({ auth, match }) => {
+  const { itemId, definitionId } = match.params;
+  const loginHandler = (itemId && definitionId) ? () => auth.loginAndLike(itemId, definitionId) : auth.login;
+  const buttonMessage = (itemId && definitionId) ? 'Login and Like' : 'Login';
```
Comment (NFabrizio) [inline src/Login.js]: Nice!

File: src/utils/Auth.js [hunk 5]
```diff
@@ -25,25 +26,41 @@ export default class Auth {
...
     let expiresAt = JSON.stringify((authResult.expiresIn * 1000) + new Date().getTime());
     localStorage.setItem('access_token', authResult.accessToken);
     localStorage.setItem('id_token', authResult.idToken);
     localStorage.setItem('expires_at', expiresAt);
     localStorage.setItem('user_id', authResult.idTokenPayload.sub);
     // navigate to the home route
+
+    if (search) {
+      const itemIdIndex = search.indexOf('&');
+      const defIdIndex = itemIdIndex + 14;
```
Comment (NFabrizio) [inline src/utils/Auth.js]: Why 14?

File: src/AcronymLike.js (hunk 3 above)
Comment (tsrahm) [reply src/AcronymLike.js]: Just my ocd so they match the order where they're declared.

File: src/utils/Auth.js (hunk 2 above)
Comment (tsrahm) [reply src/utils/Auth.js]: Good idea. Removed loginAndLike().

File: src/App.js (hunk 1 above)
Comment (tsrahm) [reply src/App.js]: Updated

File: src/utils/Auth.js (hunk 5 above)
Comment (tsrahm) [reply src/utils/Auth.js]: Yeah, I found a better method.
